docs/content/docs/test_framework/Mocking.md

```diff
@@ -5,11 +5,73 @@ type: docs
 
 # Mocking in tests
 
+To use a specific state you need to find a suitable `@Rule` or annotation.\
+Thus, test runner knows how to prepare it for the test ([more details]({{< relref "#writing-custom-mocks-for-tests">}})).
```

Comment:
i have a feeling that `#writing-custom-mocks-for-tests` should be on top of this page

Reply:
It's a tutorial and must cover popular use cases as simple as possible.

The explanation of implementation is verbose and needed only when I need to add a custom one.
Maybe it's better to extract it to another page.